language-design.md

```diff
@@ -17,6 +17,25 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+Our first quote:
+
+> In a race, a small language with warts will beat a well designed
+> language because users will not wait for the right thing; they will use the language that
+> is quick and cheap, and put up with the warts. [Steele 1998]
+
+We thought this was an interesting idea - Steele’s argument against small languages was straightforward, but we were not as convinced by this. There are certainly situations where you’ll need this kind of speed in putting out a language, but for a DSL there is not necessarily a big market of competitors that you need to beat to the punch. If there are no competitors then we thought it might make more sense to be slower and get rid of some of the warts before releasing a language. We also thought the idea of warts was interesting, and agreed with the idea that incremental design starting with a ‘warty’ language is a good solution for a lot of languages. However, this depends on the severity of the warts because some problems are less easily fixable after the language is released.
+
+Our second quote:
+
+> Not all programming languages have a way to code definitions, but most do. (Those that do not >are for wimps.) [Steele 1998]
+
+It struck us how Steele characterized languages without definitions as necessarily a bad thing, and with some discussion we realized how this is especially incorrect for DSLs. While we agreed that general-purpose languages will certainly need some way to give definitions, it may be more than the user needs for many DSLs. We thought about the DSLs we had seen and many do not have any capability for user-made definitions, for instance ABC notation (which Josh saw in a peer review). Any DSLs that function like APIs will also lack this capability because you do not need to define new commands or functions to interface properly with an API.
+
+Our third quote:
+
+> You can always add things later, but you can't take them away. [Bloch 2007 ~20 mins?]
+
+To us, this spoke directly to the fact that user-centered design is at the top of the priority list for a programming language. If there were no users, we could do whatever we wanted to the languages we design. However, if we put out a language or an API and then take away a feature that users have become accustomed to, they will naturally complain. They would be right to complain about this because if there is some feature being offered, then they should be able to become accustomed to using that feature. The set of available features should only ever grow, not shrink.
 
```

Comment:
Users are definitely frustrated by the absence of a feature they used to have. That being said, if a truly horrendous feature exists, it probably is still a good idea to get rid of it.